domain/application/service/service.go

```diff
@@ -484,6 +484,54 @@ func (s *WatchableService) WatchApplicationExposed(ctx context.Context, name str
 	)
 }
 
+// WatchUnitForUniterChanged watches for some specific changes to the unit with
+// the given name. The watcher will emit a notification when there is a change to
+// the unit's inherent properties, it's subordinates or it's resolved mode.
+//
+// If the unit does not exist an error satisfying [applicationerrors.UnitNotFound]
+// will be returned.
+//
+// These tables are selected since they provide coverage for the events the uniter
+// watches for using the WatchUnit agent facade method.
+//
+// TODO(jack-w-shaw): This watcher only exists to maintain backwards compatibility
+// with 3.6. Ideally, the facade would supply watchers for specific changes, and
+// the uniter would pick and chose the watchers it needs for each specific purpose.
+// However, we cannot do this without a facade bump, which cannot be done at the
+// moment. When we can bump the facade, we should expose separate watchers for
+// the unit life, it's subordinates, the resolve mode, etc.
+func (s *WatchableService) WatchUnitForUniterChanges(ctx context.Context, unitName coreunit.Name) (watcher.NotifyWatcher, error) {
+	uuid, err := s.GetUnitUUID(ctx, unitName)
+	if err != nil {
+		return nil, errors.Errorf("getting ID of unit %s: %w", unitName, err)
+	}
+
+	unitNamespace, principalNamespace, resolvedNamespace := s.st.NamespaceForWatchUnitForUniterChanges()
+	return s.watcherFactory.NewNotifyWatcher(
+		eventsource.PredicateFilter(
+			unitNamespace,
+			changestream.All,
+			func(s string) bool {
```

Comment:
This predicate could be better as a high order function.

```
func always(uuid coreunit.UUID) func(string) bool {
...
}
```

Reply:
I have pushed a commit on top of this PR where I give this a shot, although I'm not convinced this is a useful abstraction. Let me know your thoughts